src/util/recentlyEditedTree.js

```diff
@@ -0,0 +1,214 @@
+import _ from 'lodash'
+import { timeDifference, contextOf, equalArrays, timestamp, pathToContext } from '../util.js'
+import { produce } from 'immer'
+
+const EDIT_TIME_MAX = 7200 // time diff limit in second for replacing descendants by ancestor
+
+/**
+   * finds tree deepeset common subcontext for given a given path
+   * @param {Object} tree nested object representing tree
+   * @param {string[]} context array of string representing path
+   * @returns {string[]} common subcontext
+   */
+const findTreeDeepestSubcontext = (tree, context, index = 0) => {
```

Comment:
It will return root tree itself . It is because even when new context that has no ancestor at all in the tree at is given it could be a possible new leaf node. For example:

-A
-B

For example if we have these thoughts in the main tree, recently edited list will be

-B  (Since A and B have direct siblings relation , A and B will be merged to be just B)

Now when C is added as `subthought` of A then A.C will have no common path in the recently edited tree at all. But it is a possible case of new thought being added to the recently edited tree (A.C and B will must merge to become A.C ). Therefore we need return the main tree as the `commonNode`.